src/Controller/BlogController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Article;
+use App\Entity\Tag;
+use Doctrine\ORM\EntityManagerInterface;
+use FPN\TagBundle\Entity\TagManager;
+use Knp\Component\Pager\PaginatorInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+
+/**
+ * Blog controller.
+ */
+class BlogController extends AbstractController
+{
+    private $knpPaginator;
+
+    private $tagManager;
+
+    private $doctrine;
+
+    public function __construct(PaginatorInterface $knpPaginator, TagManager $tagManager, EntityManagerInterface $doctrine)
+    {
+        $this->knpPaginator = $knpPaginator;
+        $this->tagManager = $tagManager;
+        $this->doctrine = $doctrine;
+    }
+
+    /**
+     * @Route("/", name="blog_home")
+     */
+    public function indexAction(Request $request)
+    {
+        $query = $this->getDoctrine()->getRepository('App:Article')->getQuery();
```

Comment:
Why not create service for this ?

Reply:
First I see nothing wrong with it. But also it was like this before: https://github.com/afsy/website/blob/master/src/Afsy/Bundle/FrontBundle/Controller/BlogController.php#L27